storage/aptosdb/src/lib.rs

```diff
@@ -638,22 +638,21 @@ impl AptosDB {
     fn get_events_by_event_key(
         &self,
         event_key: &EventKey,
-        start_seq_num: u64,
+        start_seq_num: Option<u64>,
         order: Order,
         limit: u64,
         ledger_version: Version,
     ) -> Result<Vec<EventWithVersion>> {
         error_if_too_many_requested(limit, MAX_LIMIT)?;
-        let get_latest = start_seq_num == u64::MAX;
 
-        let cursor = if get_latest {
+        let cursor = if let Some(seq_num) = start_seq_num {
+            seq_num
+        } else {
```

Comment:
this is the real change

Reply:
don't you need to set the cursor to something else if it is ascending?

wouldn't this code return only 1 element if start_seq_num=None and order::Ascending is passed?

Reply:
right, fxxk, ...

guess no one is using Order::Descending any more, we should just eliminate `Order`?

Reply:
Leader_reputation is :)

But it can reverse as well :)

Reply:
:trollface:

Reply:
Sounds like a good idea since the Rest API using the current form of the DB API actually reverses twice.

But it requires a bit more than throwing in some quick & dirty PR in 10 mins. 

I'd suggest you go with the dirty reverse twice route for you PR for now and I follow up later to make it cleaner? I think it's better than calling get_latest_seq_num from the Rest API. What do you think?

Reply:
yeah, I think current state of #2582 is not that bad :) so if you can review/accept, we can proceed with that for now